src/test/java/com/epam/test_generator/controllers/TestResultControllerTest.java

```diff
@@ -0,0 +1,251 @@
+package com.epam.test_generator.controllers;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.epam.test_generator.dto.RawSuitResultDTO;
+import com.epam.test_generator.dto.TestResultDTO;
+import com.epam.test_generator.entities.Status;
+import com.epam.test_generator.services.TestResultService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.Authentication;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestResultControllerTest {
+
+    private static final long SIMPLE_PROJECT_ID = 0L;
+    private static final Integer FROM = 2;
+    private static final Integer TO = 3;
+    private static final Integer NEGATIVE = -5;
+    private static final SimpleDateFormat format = new SimpleDateFormat("dd/mm/yyyy");
+    private Date FROM_DATE;
+    private Date TO_DATE;
+
+    private ObjectMapper mapper = new ObjectMapper();
+    private MockMvc mockMvc;
+    private TestResultDTO testResultDTO;
+    private List<TestResultDTO> testResultDTOS;
+    private List<RawSuitResultDTO> rawSuitResultDTOS;
+
+    @Mock
+    private TestResultService testResultService;
+
+    @InjectMocks
+    private TestResultController testResultController;
+
+    @Before
+    public void setup() throws ParseException {
+        mockMvc = MockMvcBuilders.standaloneSetup(testResultController)
+            .setControllerAdvice(new GlobalExceptionController()).build();
+
+        testResultDTOS = new ArrayList<>();
+        FROM_DATE = format.parse("26/02/2018");
+        TO_DATE = format.parse("27/02/2018");
+
+        testResultDTO = new TestResultDTO();
```

Comment:
Are you serious? You have 7 equal objects which only has different date and you created each of them. Youo can create a private method which will return or will add a new TestResultDTO to your testResultDTOS with different dates.

```java
    private void testResult(Date date) {
        testResultDTO = new TestResultDTO();
        testResultDTO.setAmountOfFailed(0);
        testResultDTO.setAmountOfPassed(1);
        testResultDTO.setAmountOfSkipped(0);
        testResultDTO.setDate(date);
        testResultDTO.setDuration(0);
        testResultDTO.setExecutedBy("User Userovich");
        testResultDTO.setStatus(Status.PASSED);
        testResultDTO.setSuits(Collections.emptyList());
        testResultDTOS.add(testResultDTO);
    }
```
And in your code you can only call this method:
```java
        testResult(new Date(1519618194914L));
        testResult(new Date(1519628194914L));
        testResult(FROM_DATE);
        testResult(FROM_DATE);
        testResult(new Date(1519668194914L));
        testResult(TO_DATE);
        testResult(new Date(1519688194914L));
```